crates/ruff_wasm/src/lib.rs

```diff
@@ -298,9 +301,14 @@ impl<'a> ParsedModule<'a> {
         })
     }
 
-    fn format(&self) -> FormatResult<Formatted<PyFormatContext>> {
+    fn format(&self, settings: &Settings) -> FormatResult<Formatted<PyFormatContext>> {
         // TODO(konstin): Add an options for py/pyi to the UI (2/2)
-        let options = PyFormatOptions::from_source_type(PySourceType::default());
+        let options = PyFormatOptions::from_source_type(PySourceType::default())
+            .with_preview(match settings.preview {
+                PreviewMode::Disabled => ruff_python_formatter::PreviewMode::Disabled,
+                PreviewMode::Enabled => ruff_python_formatter::PreviewMode::Enabled,
+            })
+            .with_line_width(LineWidth::from(NonZeroU16::from(settings.line_length)));
```

Comment:
we should add an `PyFormatOptions::from_settings(source_type: PySourceType, settings: &Settings` or similar

Reply:
Agree. The challenge is where to put this because `Setting`s is defined in `ruff`. The only real option is `ruff_workspace` but we should probably split settings into smaller sub-setting objects (formatter, linter, etc).

Reply:
it would be nice to have resolved setting outside of `ruff`. I'm not sure how to handle the rules best, maybe move the registry out or have rule-resolved settings that "subclass" the general settings.

Reply:
My idea is to have a `LinterSettings` object that we can resolve from `Settings`. Similarly, we have `PyFormatOptions` that we can resolve from `Settings`. Or we have smaller sub-settings but yeah, not entirely sure how to structure this without having to clone a whole lot of data.